extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/DashMpdParser.java

```diff
@@ -0,0 +1,221 @@
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.annotation.Nonnull;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+/*
+ * Created by Christian Schabesberger on 02.02.16.
+ *
+ * Copyright (C) Christian Schabesberger 2016 <chris.schabesberger@mailbox.org>
+ * DashMpdParser.java is part of NewPipe Extractor.
+ *
+ * NewPipe Extractor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * NewPipe Extractor is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with NewPipe Extractor. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+public class DashMpdParser {
+    private DashMpdParser() {
+    }
+
+    public static class DashMpdParsingException extends ParsingException {
+
+        DashMpdParsingException(final String message, final Exception e) {
+            super(message, e);
+        }
+    }
+
+    public static class Result {
+        private final List<VideoStream> videoStreams;
+        private final List<VideoStream> videoOnlyStreams;
+        private final List<AudioStream> audioStreams;
+
+
+        public Result(final List<VideoStream> videoStreams,
+                      final List<VideoStream> videoOnlyStreams,
+                      final List<AudioStream> audioStreams) {
+            this.videoStreams = videoStreams;
+            this.videoOnlyStreams = videoOnlyStreams;
+            this.audioStreams = audioStreams;
+        }
+
+        public List<VideoStream> getVideoStreams() {
+            return videoStreams;
+        }
+
+        public List<VideoStream> getVideoOnlyStreams() {
+            return videoOnlyStreams;
+        }
+
+        public List<AudioStream> getAudioStreams() {
+            return audioStreams;
+        }
+    }
+
+    // TODO: Make this class generic and decouple from YouTube's ItagItem class.
+
+    /**
+     * Will try to download and parse the DASH manifest (using {@link StreamInfo#getDashMpdUrl()}),
+     * adding items that are listed in the {@link ItagItem} class.
+     * <p>
+     * It has video, video only and audio streams.
+     * <p>
+     * Info about DASH MPD can be found here
+     *
+     * @param dashMpdUrl URL to the DASH MPD
```

Comment:
> Link to the DASH MPD explainer

I will use `the URL of the DASH MPD file` because if we use `link`, we lost precision of what adress links are supported by the class.

Also, `file` seems to be really more appropriated here than `explainer`.